sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/ObjectBatch.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.amqp.implementation.ErrorContextProvider;
+import com.azure.core.amqp.implementation.TracerProvider;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.experimental.serializer.ObjectSerializer;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Objects;
+
+/**
+ * A class for aggregating Java objects into a single, size-limited, batch. Objects are serialized into EventData
+ * objects and are added to the batch.  It is treated as a single message when sent to the Azure Event Hubs service.
+ *
+ * @param <T> type of objects in the batch.  Multi-type batches are not permitted.
+ */
+public final class ObjectBatch<T> extends EventDataBatchBase {
+    private final ClientLogger logger = new ClientLogger(ObjectBatch.class);
+    private final Class<T> batchType;
+    private final ObjectSerializer objectSerializer;
+
+    ObjectBatch(int maxMessageSize, String partitionId, String partitionKey, Class<T> batchType,
+                    ErrorContextProvider contextProvider, TracerProvider tracerProvider,
+                    ObjectSerializer objectSerializer, String entityPath, String hostname) {
+        super(maxMessageSize, partitionId, partitionKey, contextProvider, tracerProvider, entityPath, hostname);
+        this.batchType = Objects.requireNonNull(batchType, "'batchType' cannot be null.");
+        this.objectSerializer = Objects.requireNonNull(objectSerializer, "'objectSerializer' cannot be null.");
```

Comment:
Would it be better to check this is non-null when validating batch options or could this be created from other contexts?

Reply:
Added a check in object batch creation.